tensorflow/python/ops/nn_ops.py

```diff
@@ -1109,7 +1109,7 @@ def dropout(x, keep_prob, noise_shape=None, seed=None, name=None):
   """
   with ops.name_scope(name, "dropout", [x]) as name:
     x = ops.convert_to_tensor(x, name="x")
-    if isinstance(keep_prob, float) and not 0 < keep_prob <= 1:
+    if (isinstance(keep_prob, int) or isinstance(keep_prob, float)) and not 0 < keep_prob <= 1:
       raise ValueError("keep_prob must be a scalar tensor or a float in the "
```

Comment:
It seems that the problem prevails when long ints are passed, e.g. 0L.

If you can afford an additional import,

import numbers
isinstance(keep_prob, numbers.Real)

appears to be the proper solution.

http://stackoverflow.com/questions/4541155/check-if-a-number-is-int-or-float



On 25 August 2016 at 06:44, Derek Murray <notifications@github.com> wrote:

> In tensorflow/python/ops/nn_ops.py
> <https://github.com/tensorflow/tensorflow/pull/4015#discussion_r76179387>:
>
> > @@ -1109,7 +1109,7 @@ def dropout(x, keep_prob, noise_shape=None, seed=None, name=None):
> >    """
> >    with ops.name_scope(name, "dropout", [x]) as name:
> >      x = ops.convert_to_tensor(x, name="x")
> > -    if isinstance(keep_prob, float) and not 0 < keep_prob <= 1:
> > +    if (isinstance(keep_prob, int) or isinstance(keep_prob, float)) and not 0 < keep_prob <= 1:
> >        raise ValueError("keep_prob must be a scalar tensor or a float in the "
>
> Sorry for the drive-by, but isinstance(keep_prob, (float, int)) would be
> more idiomatic (and compact!) here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/4015/files/1028c83d9fd338149c305604c1a89e93ff02812b#r76179387>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAuK0fLQLW9PtD5QkyQjYkFDJk5Mop3Kks5qjR2xgaJpZM4Jr7f_>
> .
>